connection_manager.go

```diff
@@ -308,6 +308,15 @@ func (cm *connectionManager) expireConnections() {
 
 			logDebug("[connectionManager]", "(%v) expired %d connections.", cm, expiredCount)
 
+			for cm.connectionCount < cm.minConnections {
```

Comment:
I also wasn't sure of the best way you would like to test this. I haven't succeeded in getting the integration tests running locally. I think I could write a pretty straightforward unit test for it before it's merged in as well.